### PR TITLE
[Improvement]: Ingress configration for ams when deploying with Helm.

### PR DESCRIPTION
**Pull request by baiyangtx**
## Why are the changes needed?

Add ingress configuration in Chart.  Make Helm user easier to access AMS Dashborad.

By default, applications within a Kubernetes cluster with Ingress are not accessible from outside the cluster. But when an IngressClass is installed in the Kubernetes cluster, it is possible to access the installed application pages on the cluster from outside the cluster by creating Ingress resources.

More information about ingress refer here: https://kubernetes.io/docs/concepts/services-networking/ingress/

## Brief change log

- Add Ingress config in Chart

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [X] Add screenshots for manual tests if appropriate

- [ ] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (yes )
- If yes, how is the feature documented? (not documented)

### CODE REVIEW

**Comment by czy006** [comment]
When I try to value.yaml set ingress config,when "helm dependency build" before install helm chart

values.yaml

```
ingress: 
  enabled: true
  hostname: amoro.ingress.minikube.local
```
It have something problem

- install deployment service account can't create
- set ingress annotations can't merge have error msg

Error: INSTALLATION FAILED: template: amoro/templates/amoro-ingress.yaml:27:22: executing "amoro/templates/amoro-ingress.yaml" at <include "common.tplvalues.merge" (dict "values" (list .Values.ingress.annotations .Values.commonAnnotations) "context" .)>: error calling include: template: no template "common.tplvalues.merge" associated with template "gotpl"



image

Env：

- Docker Desktop Version: 4.23.0 (120376)
- K8S Version: 1.27.2
- Helm Version: 3.12.3

cc @baiyangtx

File: charts/amoro/templates/amoro-ingress.yaml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+apiVersion: {{ template "common.capabilities.ingress.apiVersion" . }}
+kind: Ingress
+metadata:
+  name:  {{ include "common.names.fullname" . }}
+  namespace: {{ include "common.names.namespace" . | quote }}
+  labels: {{- include "common.labels.standard" ( dict "customLabels" .Values.commonLabels "context" $ ) | nindent 4 }}
+    app.kubernetes.io/component: ingress
+  {{- if or .Values.ingress.annotations .Values.commonAnnotations }}
+  {{- $annotations := include "common.tplvalues.merge" ( dict "values" ( list .Values.ingress.annotations .Values.commonAnnotations ) "context" . ) }}
+  annotations: {{- include "common.tplvalues.render" ( dict "value" $annotations "context" $) | nindent 4 }}
```
Comment (czy006) [inline charts/amoro/templates/amoro-ingress.yaml:28]: when I install it set 
```
  annotations:
    kubernetes.io/ingress.class: nginx
```
that have error helm merge msg: _Error: INSTALLATION FAILED: template: amoro/templates/amoro-ingress.yaml:27:22: executing "amoro/templates/amoro-ingress.yaml" at <include "common.tplvalues.merge" (dict "values" (list .Values.ingress.annotations .Values.commonAnnotations) "context" .)>: error calling include: template: no template "common.tplvalues.merge" associated with template "gotpl"_

**Review by czy006 — CHANGES_REQUESTED** [review]

**Comment by XBaith** [comment]
I tried installing the chart with the following values, but it doesn't seem to work:
```
ingress: 
  enabled: true
  hostname: amoro.ingress.minikube.local
```

File: charts/amoro/Chart.lock
[the remark this answers is not in the document]
Comment (baiyangtx) [reply charts/amoro/Chart.lock:1]: This file is generated by Helm to record the version information of dependent libraries.

**Comment by baiyangtx** [comment]
> I tried installing the chart with the following values, but it doesn't seem to work:
> 
> ```
> ingress: 
>   enabled: true
>   hostname: amoro.ingress.minikube.local
> ```

Do you have an default IngressClass implementation in your cluster ?

File: charts/amoro/templates/amoro-ingress.yaml (hunk 1 above)
Comment (baiyangtx) [reply charts/amoro/templates/amoro-ingress.yaml:28]: I have fixed by upgrading `bitnamicharts/common` to version 2.11.1

**Comment by baiyangtx** [comment]
> When I try to value.yaml set ingress config,when "helm dependency build" before install helm chart
> 
> values.yaml
> 
> ```
> ingress: 
>   enabled: true
>   hostname: amoro.ingress.minikube.local
> ```
> 
> It have something problem
> 
>     * install deployment service account can't create
> 
>     * set ingress annotations can't merge have error msg
> 
> 
> Error: INSTALLATION FAILED: template: amoro/templates/amoro-ingress.yaml:27:22: executing "amoro/templates/amoro-ingress.yaml" at <include "common.tplvalues.merge" (dict "values" (list .Values.ingress.annotations .Values.commonAnnotations) "context" .)>: error calling include: template: no template "common.tplvalues.merge" associated with template "gotpl"
> image
> 
> Env：
> 
>     * Docker Desktop Version: 4.23.0 (120376)
> 
>     * K8S Version: 1.27.2
> 
>     * Helm Version: 3.12.3
> 
> 
> cc @baiyangtx

I have fixed the ANNOTATION problems and account problems.

**Comment by HuangFru** [comment]
Env:
- Docker:24.0.2-rd
- Minikube:v1.31.2

values.yaml
```
ingress: 
  enabled: true
  hostname: amoro.ingress.minikube.local
```
process:
```
➜  helm install amoro ../amoro -f values.yaml 
NAME: amoro
LAST DEPLOYED: Mon Sep 25 16:03:12 2023
NAMESPACE: default
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
The Amoro Helm chart has been installed!

If you want to go ams web ui,you can use this cmd forward to local:


  http://amoro.ingress.minikube.local/


Additional Resources Docs

================================================================================================================

        * Documentation: https://amoro.netease.com/docs/latest/
        * Version build Info : https://github.com/NetEase/amoro/releases/tag/v0.5.0

================================================================================================================

```
```
➜  kubectl get ingress
NAME    CLASS    HOSTS                          ADDRESS   PORTS   AGE
amoro   <none>   amoro.ingress.minikube.local             80      19s
➜  kubectl describe ingress
Name:             amoro
Labels:           app.kubernetes.io/component=ingress
                  app.kubernetes.io/instance=amoro
                  app.kubernetes.io/managed-by=Helm
                  app.kubernetes.io/name=amoro
                  app.kubernetes.io/version=0.5.0
                  helm.sh/chart=amoro-0.1.0
Namespace:        default
Address:          
Ingress Class:    <none>
Default backend:  <default>
Rules:
  Host                          Path  Backends
  ----                          ----  --------
  amoro.ingress.minikube.local  
                                /   amoro-rest:rest (10.244.0.5:1630)
Annotations:                    meta.helm.sh/release-name: amoro
                                meta.helm.sh/release-namespace: default
Events:                         <none>
```
It does not work for me, I can not visit AMS web UI via ' http://amoro.ingress.minikube.local/'. It seems that I lost the address in the ingress.

**Comment by baiyangtx** [comment]
> Env:
> 
>     * Docker:24.0.2-rd
> 
>     * Minikube:v1.31.2
> 
> 
> values.yaml
> 
> ```
> ingress: 
>   enabled: true
>   hostname: amoro.ingress.minikube.local
> ```
> 
> process:
> 
> ```
> ➜  helm install amoro ../amoro -f values.yaml 
> NAME: amoro
> LAST DEPLOYED: Mon Sep 25 16:03:12 2023
> NAMESPACE: default
> STATUS: deployed
> REVISION: 1
> TEST SUITE: None
> NOTES:
> The Amoro Helm chart has been installed!
> 
> If you want to go ams web ui,you can use this cmd forward to local:
> 
> 
>   http://amoro.ingress.minikube.local/
> 
> 
> Additional Resources Docs
> 
> ================================================================================================================
> 
>         * Documentation: https://amoro.netease.com/docs/latest/
>         * Version build Info : https://github.com/NetEase/amoro/releases/tag/v0.5.0
> 
> ================================================================================================================
> ```
> 
> ```
> ➜  kubectl get ingress
> NAME    CLASS    HOSTS                          ADDRESS   PORTS   AGE
> amoro   <none>   amoro.ingress.minikube.local             80      19s
> ➜  kubectl describe ingress
> Name:             amoro
> Labels:           app.kubernetes.io/component=ingress
>                   app.kubernetes.io/instance=amoro
>                   app.kubernetes.io/managed-by=Helm
>                   app.kubernetes.io/name=amoro
>                   app.kubernetes.io/version=0.5.0
>                   helm.sh/chart=amoro-0.1.0
> Namespace:        default
> Address:          
> Ingress Class:    <none>
> Default backend:  <default>
> Rules:
>   Host                          Path  Backends
>   ----                          ----  --------
>   amoro.ingress.minikube.local  
>                                 /   amoro-rest:rest (10.244.0.5:1630)
> Annotations:                    meta.helm.sh/release-name: amoro
>                                 meta.helm.sh/release-namespace: default
> Events:                         <none>
> ```
> 
> It does not work for me, I can not visit AMS web UI via ' http://amoro.ingress.minikube.local/'. It seems that I lost the address in the ingress.

You have not been installed  an IngressClass on you k8s Cluster.


run `kubectl get ingressclass`

**Comment by HuangFru** [comment]
For macOS M series chips to use the ingress addons in minikube, according to https://github.com/kubernetes/minikube/issues/13510:
- `minikube tunnel` is needed.
- Mapping your domain to 127.0.0.1 instead of the output of `minikube ip` in the hosts file.

After these things, this PR works fine for me.

**Review by HuangFru — APPROVED** [review]
LGTM.

**Review by czy006 — APPROVED** [review]

**Comment by czy006** [comment]
LGTM.

**Comment by baiyangtx** [comment]
> For macOS M series chips to use the ingress addons in minikube, according to [kubernetes/minikube#13510](https://github.com/kubernetes/minikube/issues/13510):
> 
>     * `minikube tunnel` is needed.
> 
>     * Mapping your domain to 127.0.0.1 instead of the output of `minikube ip` in the hosts file.
> 
> 
> After these things, this PR works fine for me.

Thanks your testing. This is helpful for using on Mac, and I will add this additional configuration to the documentation.

**Review by majin1102 — APPROVED** [review]
LGTM

**Review by shidayang — APPROVED** [review]

**Pull request merged**